### PR TITLE
consolidates ingester pipeline as macrorepo

### PR DESCRIPTION
**Pull request by kotharironak**
Additional changes require on top of 
- https://github.com/hypertrace/hypertrace-ingester/pull/14/
- https://github.com/hypertrace/hypertrace-ingester/pull/15/

Validated,
1. by creating subtree locally
Locally, I have added all modules as a subtree
```
for MODULE_NAME in span-normalizer raw-spans-grouper hypertrace-trace-enricher hypertrace-view-generator; do
  git config -f .gitmodules --remove-section submodule.$MODULE_NAME
  git add .gitmodules
  git rm --cached $MODULE_NAME
  rm -rf .git/modules/$MODULE_NAME
  git commit -m "Removed submodule $MODULE_NAME per #8"
  rm -rf $MODULE_NAME
  git subtree add --prefix=${MODULE_NAME} https://github.com/hypertrace/${MODULE_NAME}.git main
done
```

2. incorporating changes from ` https://github.com/hypertrace/hypertrace-ingester/pull/15/`

### CODE REVIEW

File: .circleci/helm.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+script=$0
+subcommand=$1; shift
+
+case "$subcommand" in
+  validate)
+    for SUB_PROJ_DIR in $SUB_PROJECTS_DIRS; do
+      cd $SUB_PROJ_DIR
+      echo "*******"
+      echo "validating charts for:$(pwd)"
```
Comment (jcchavezs) [inline .circleci/helm.sh:17]:
```suggestion
      echo "Validating charts for dir \"$(pwd)\""
```

Wrapping the value with quotes is really helpful when having weird things like empty string.

File: .circleci/helm.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+cd $ROOT_PROJECT_DIR
+SUB_PROJECTS_DIRS=$(find . -iname "helm" | sed 's/\(.*\)\/.*/\1/')
+
+script=$0
+subcommand=$1; shift
+
+case "$subcommand" in
+  validate)
+    for SUB_PROJ_DIR in $SUB_PROJECTS_DIRS; do
+      cd $SUB_PROJ_DIR
```
Comment (jcchavezs) [inline .circleci/helm.sh:15]: Do we need to cd the `$SUB_PROJ_DIR`, can't we just run

```
helm dependency update $SUB_PROJ_DIR/helm/
helm lint --strict $SUB_PROJ_DIR/helm/
helm template $SUB_PROJ_DIR/helm/ --output-dir $SUB_PROJ_DIR/helm/
```

**Comment by jcchavezs** [comment]
Left a couple of comments. Overall looks great, good job! I wonder if we should commit the helm charts generated in this repository too.

File: hypertrace-ingester/build.gradle.kts [hunk 3]
```diff
@@ -25,12 +29,12 @@ dependencies {
...
   implementation("org.hypertrace.core.serviceframework:platform-metrics:0.1.8")
   implementation("org.hypertrace.core.datamodel:data-model:0.1.9")
-
-  implementation("org.hypertrace.core.spannormalizer:span-normalizer")
-  implementation("org.hypertrace.core.rawspansgrouper:raw-spans-grouper")
-  implementation("org.hypertrace.traceenricher:hypertrace-trace-enricher")
-  implementation("org.hypertrace.viewgenerator:hypertrace-view-generator")
   implementation("org.hypertrace.core.viewgenerator:view-generator-framework:0.1.14")
+
+  implementation(project(":span-normalizer:span-normalizer"))
```
Comment (adriancole) [inline hypertrace-ingester/build.gradle.kts:34]: this is broke for me locally. I will push it to the other branch and we can fix it there.

**Comment by adriancole** [comment]
lets work on #15 as it is hard to sync very different branches. note it is currently broken due to 

```
* What went wrong:
Could not determine the dependencies of task ':hypertrace-ingester:syncJavaApplicationDockerContext'.
> Could not resolve all task dependencies for configuration ':hypertrace-ingester:runtimeClasspath'.
   > Could not resolve project :raw-spans-grouper:raw-spans-grouper.
     Required by:
```

**Pull request closed without merging**